### PR TITLE
feat: Add Fields impl for Link

### PR DESCRIPTION
**Pull request by alekzvik**
## Closes

-  #501 
## Description
Add implementation of `Field` trait for `Link`.

## Checklist

Delete any checklist items that do not apply (e.g. if your change is minor, it may not require documentation updates).

- [x] Unit tests
- [ ] Documentation, including doctests
- [x] Git history is linear
- [x] Commit messages are descriptive
- [x] (optional) Git commit messages follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Code is formatted (`cargo fmt`)
- [x] `cargo test`
- [x] Changes are added to the CHANGELOG

### CODE REVIEW

**Comment by alekzvik** [comment]
Disclaimer:
I am new to rust, so I picked an issue and tried to do it. 
Please let me know if something can be improved.

**Review by gadomski — CHANGES_REQUESTED** [review]
Awesome, thanks for the contribution! As of https://github.com/stac-utils/stac-rs/pull/507, I think we could just `#[derive(Fields)]` on the `Links` structure, rather than manually implementing it. See https://github.com/stac-utils/stac-rs/blob/533a8b35072a1ae14cebbf3a0eaae759de2fc725/crates/core/src/collection.rs#L25 for an example.

File: crates/types/src/link.rs [hunk 1]
```diff
@@ -701,6 +701,25 @@ mod tests {
         assert!(link.title.is_none());
     }
 
+    mod fields {
```
[the remark this answers is not in the document]
Comment (gadomski) [reply crates/types/src/link.rs:704]: Probably not

**Comment by gadomski** [comment]
> I am a bit lost with the CHANGELOG.md though. The change I made is in the stac_types crate. Do we need a separate CHANGELOG.md for it? If yes, did you use any tool to generate the first version?

Yeah, `stac-types` is new, and I _think_ won't be used by most folks. We re-export it in `stac` so it's fine to update the changelog there, thanks!

**Review by gadomski — APPROVED** [review]

**Pull request merged**